src/flamenco/vm/fd_vm_interp_core.c

```diff
@@ -163,7 +163,7 @@
   shadow[ frame_cnt ].r9 = reg[9];                                                                          \
   shadow[ frame_cnt ].pc = pc;                                                                              \
   if( FD_UNLIKELY( ++frame_cnt>=frame_max ) ) goto sigstack; /* Note: untaken branches don't consume BTB */ \
-  reg[10] += FD_VM_STACK_FRAME_SZ + FD_VM_STACK_GUARD_SZ
+  reg[10] += FD_VM_STACK_FRAME_SZ + ( FD_FEATURE_ACTIVE( vm->instr_ctx->slot_ctx, bpf_account_data_direct_mapping ) ? 0 : FD_VM_STACK_GUARD_SZ )
```

Comment:
can we move this into a variable and put it at the top of fd_vm_interp_core.c or fd_vm_interp.c?